pkgs/applications/terminal-emulators/mlterm/default.nix

```diff
@@ -98,6 +99,7 @@ let
       commaSepList = lib.concatStringsSep "," (builtins.attrNames (lib.filterAttrs (n: v: v) attrset));
     in
     lib.withFeatureAs (commaSepList != "") featureName commaSepList;
+  stdenv = if args.stdenv.cc.isGNU then args.gcc13Stdenv else args.stdenv;
```

Comment:
It does not look like anything more elaborate than `-Wno-error=incompatible-pointer-types` would be required to fix compilation, which is far preferable to pinning the old GCC. However, upstream has already fixed these issues in e.g. https://github.com/arakiken/mlterm/commit/3d38b723e0e4a6dd434af2d49aca53890982a828 and https://github.com/arakiken/mlterm/commit/821a5563cc9794bb344bdbc83207a173c1a2b83f. So let’s just apply the patches instead.